### PR TITLE
Adding json export

### PR DESCRIPTION
**Pull request by Deadloock**
Adding methods for exporting log data into JSON file

## Description
This PR includes methods for exporting log data (LogData) into JSON file.

## Related Issue
https://github.com/adobe/log-parser/issues/173

## Motivation and Context
We have options to export LogData into CSV and HTML, with this code change we can now export data into JSON as well

## How Has This Been Tested?
This is tested by creating new unit test testExportDataToJSON

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/test/java/com/adobe/campaign/tests/logparser/core/LogDataTest.java [hunk 1]
```diff
@@ -1472,4 +1472,40 @@ public void testFetchFirst() {
...
+
+        int l_nrOfEntries = l_logData.getEntries().keySet().size();
+        assertThat("The LogData needs to have been generated", l_nrOfEntries, Matchers.greaterThan(0));
+
+        File l_exportedFile = l_logData.exportLogDataToJSON();
+
+        try {
+            assertThat("We successfully created the file", l_exportedFile, notNullValue());
+            assertThat("We successfully created the file", l_exportedFile.exists());
+            assertThat("We successfully created the file correctly", l_exportedFile.isFile());
```
Comment (baubakg) [inline src/test/java/com/adobe/campaign/tests/logparser/core/LogDataTest.java:1506]: I suggest that you check the file is not empty. 

Ideally, I would import it and see if the map has the same number of ntries

File: src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java [hunk 2]
```diff
@@ -490,6 +491,74 @@ public File exportLogDataToHTML(Collection<String> in_headerSet, String in_repor
...
+        File l_exportFile = LogParserFileUtils.createNewFile(in_jsonFileName);
+        List<Map<String, String>> JSONlist = new ArrayList<>();
+
+        try {
+            for (StdLogEntry lt_entry : this.getEntries().values()) {
+                Map lt_values = lt_entry.fetchValueMap();
+                JSONlist.add(lt_values);
+            }
+        }
+        catch (Exception e) {
```
Comment (baubakg) [inline src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java:546]: Doesn't seem right. I am not sure catching "excption" is a good idea

File: src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java [hunk 3]
```diff
@@ -490,6 +491,74 @@ public File exportLogDataToHTML(Collection<String> in_headerSet, String in_repor
...
+     * Exports the current LogData to an JSON file
+     *
+     * @param in_headerSet   A set of headers to be used as keys for exporting
+     * @param in_jsonFileName The file name to export
+     * @return a JSON file containing the LogData
+     * @throws LogDataExportToFileException If the file could not be exported
+     */
+    public File exportLogDataToJSON(Collection<String> in_headerSet, String in_jsonFileName) throws LogDataExportToFileException {
+        File l_exportFile = LogParserFileUtils.createNewFile(in_jsonFileName);
+        List<Map<String, String>> JSONlist = new ArrayList<>();
```
Comment (baubakg) [inline src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java:538]: A local variable should not start with capital letters.

File: src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java [hunk 4]
```diff
@@ -490,6 +491,74 @@ public File exportLogDataToHTML(Collection<String> in_headerSet, String in_repor
...
+
+        try {
+            for (StdLogEntry lt_entry : this.getEntries().values()) {
+                Map lt_values = lt_entry.fetchValueMap();
+                JSONlist.add(lt_values);
+            }
```
Comment (baubakg) [inline src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java:544]: This can be done in a stream

**Review by baubakg — CHANGES_REQUESTED** [review]

File: src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java (hunk 2 above)
Comment (Deadloock) [reply src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java:546]: Removed

File: src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java (hunk 3 above)
Comment (Deadloock) [reply src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java:538]: Variable renamed

File: src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java (hunk 4 above)
Comment (Deadloock) [reply src/main/java/com/adobe/campaign/tests/logparser/core/LogData.java:544]: Changed, so we use stream now

**Pull request merged**